eagle-xl/scripts/bh_metallicity_evolution.py

```diff
@@ -0,0 +1,93 @@
+"""
+Plots the stellar metallicity density evolution.
+"""
+import matplotlib
+
+import unyt
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sys
+
+
+from swiftsimio import load, load_statistics
+
+from swiftpipeline.argumentparser import ScriptArgumentParser
+
+arguments = ScriptArgumentParser(
+    description="Creates a metallicity density evolution plot for gas, stars and black holes."
+)
+
+snapshot_filenames = [
+    f"{directory}/{snapshot}"
+    for directory, snapshot in zip(arguments.directory_list, arguments.snapshot_list)
+]
+
+stats_filenames = [
+    f"{directory}/statistics.txt" for directory in arguments.directory_list
+]
+
+names = arguments.name_list
+output_path = arguments.output_directory
+
+plt.style.use(arguments.stylesheet_location)
+
+simulation_lines = []
+simulation_labels = []
+
+fig, ax = plt.subplots()
+
+ax.loglog()
+
+for snapshot_filename, stats_filename, name in zip(
+    snapshot_filenames, stats_filenames, names
+):
+    data = load_statistics(stats_filename)
+
+    snapshot = load(snapshot_filename)
+    boxsize = snapshot.metadata.boxsize
+    box_volume = boxsize[0] * boxsize[1] * boxsize[2]
+
+    # a, Redshift, SFR
+    scale_factor = data.a
+    redshift = data.z
+    bh_Z_mass = data.bh_z_mass
+    bh_Z_mass_density = bh_Z_mass / box_volume
```

Comment:
If here you do `bh_Z_mass_density = (bh_Z_mass / box_volume).to("Solar_Mass / Mpc**3")` you will always ensure that the value that you plot has those units, if that makes sense.